app/controllers/api/v1/recordings_controller.rb

```diff
@@ -37,6 +38,17 @@ def destroy
         render_json(status: :ok)
       end
 
+      # PUT/PATCH /api/v1/recordings/:recording_id.json
+      # Returns: { data: Array[serializable objects(recordings)] , errors: Array[String] }
+      # Does: Update the recording name.
+      def update
+        new_recording_name = recording_update_params[:name]
+        return render_json errors: [Rails.configuration.custom_error_msgs[:missing_params]], status: :bad_request if new_recording_name.blank?
+
+        RecordingsUpdater.new(recording: @recording, new_name: new_recording_name).call
```

Comment:
You also need to update the local `Recording` object

Reply:
Hmm, I didn't approach it that way.
In this case there will not be any need for a full r-sync.
This will help solve all of the issues and further simplify the tests as you have mentioned.
Thanks, I'll rectify the design!